tests/keyinput/test_basekeyparser.py

```diff
@@ -63,44 +63,26 @@ class TestSplitCount:
 
     """Test the _split_count method.
 
-    Attributes:
-        kp: The BaseKeyParser we're testing.
+    Class Attributes:
+        TESTS: list of parameters for the tests, as tuples of
+        (input_key, supports_count, expected)
     """
 
-    @pytest.fixture(autouse=True)
-    def setup(self):
-        self.kp = basekeyparser.BaseKeyParser(0, supports_count=True)
-
-    def test_onlycount(self):
+    TESTS = [
```

Comment:
Well we could also change the signature to have a single parameter labelled ``test_case`` and refer to its fields in the test respectively (e.g. ``test_case.input_key``).